### PR TITLE
Support version upgrading for rabbitmq and enhance e2e tests

### PR DESCRIPTION
**Pull request by euclidgame**

### CODE REVIEW

File: src/deps_hack/src/lib.rs [hunk 1]
```diff
@@ -107,6 +107,9 @@ pub struct ZookeeperConfig {
 #[kube(shortname = "rbmq", namespaced)]
 pub struct RabbitmqClusterSpec {
     pub replicas: i32,
+    /// Image is the name of the RabbitMQ docker image to use for RabbitMQ nodes in the RabbitmqCluster.
+    /// Must be provided together with ImagePullSecrets in order to use an image in a private registry.
+    pub image: String,
```
Comment (marshtompsxd) [inline src/deps_hack/src/lib.rs:112]: Did you copy the description from the official rabbitmq controller? We don't have an ImagePullSecrets here.

File: src/controller_examples/rabbitmq_controller/spec/reconciler.rs [hunk 2]
```diff
@@ -778,7 +778,13 @@ pub open spec fn make_rabbitmq_pod_spec(rabbitmq: RabbitmqClusterView) -> PodSpe
             seq![
                 ContainerView::default()
                 .set_name(new_strlit("setup-container")@)
-                .set_image(new_strlit("rabbitmq:3.11.10-management")@)
+                .set_image(rabbitmq.spec.image)
+                .set_resources(
```
Comment (marshtompsxd) [inline src/controller_examples/rabbitmq_controller/spec/reconciler.rs:782]: This reminds me: we should have a `resources` field for rabbitmq, like zookeeper

**Comment by marshtompsxd** [comment]
@euclidgame Can you also add the e2e test to test version upgrading for rabbitmq controller in the same PR?

File: src/controller_examples/rabbitmq_controller/spec/reconciler.rs (hunk 2 above)
Comment (euclidgame) [reply src/controller_examples/rabbitmq_controller/spec/reconciler.rs:782]: We already have one..

Comment (marshtompsxd) [reply src/controller_examples/rabbitmq_controller/spec/reconciler.rs:782]: just realize this is for the init_container, then nvm

**Comment by tianyin** [comment]
@marshtompsxd Is this caused by flakiness due to the timing?

**Comment by marshtompsxd** [comment]
I suspect so. I add a bit more waiting time before the failed test in another PR to hopefully remediate it.

**Pull request merged**